internal/contour/metrics_test.go

```diff
@@ -26,6 +26,43 @@ import (
 )
 
 func TestIngressRouteMetrics(t *testing.T) {
+	type testcase struct {
+		objs           []interface{}
+		wantIR         *metrics.RouteMetric
+		wantProxy      *metrics.RouteMetric
+		rootNamespaces []string
+	}
+
+	run := func(t *testing.T, name string, tc testcase) {
+		t.Helper()
+
+		t.Run(name, func(t *testing.T) {
+			t.Helper()
```

Comment:
Does this need called again after being called on line 37?

Reply:
Yep, we need to mark all the intermediate functions so that the reported line number is the place where we define and call the test case.